basic/payment-splitter/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+            .initial_supply_fungible(1);
+
+            // Creating the shareholders NFT Resource Definition. We will not mint any of them at this point
+            // of time as the minting and creation of these tokens happens when the `add_shareholder` function
+            // is called
+            let shareholder_def: ResourceDef = ResourceBuilder::new_non_fungible()
+                .metadata("name", "Shareholder")
+                .metadata("symbol", "shr")
+                .flags(MINTABLE | INDIVIDUAL_METADATA_MUTABLE | SHARED_METADATA_MUTABLE)
+                .badge(admin_badge.resource_address(), MAY_MINT)
+                .badge(internal_admin_badge.resource_address(), MAY_CHANGE_INDIVIDUAL_METADATA | MAY_CHANGE_SHARED_METADATA)
+                .no_initial_supply();
+
+            // Creating the payment splitter component
+            let payment_splitter: Component = Self {
+                xrd_vault: Vault::new(RADIX_TOKEN),
+                admin_badge_def: admin_badge.resource_def(),
+                internal_admin_badge: Vault::with_bucket(internal_admin_badge),
+                shareholder_def: shareholder_def,
+                number_of_shareholders: 0
+            }.instantiate();
+
+            // Returning the PaymentSplitter component and the admin badge
+            return (payment_splitter, admin_badge);
+        }
+
+        /// Adds a shareholder to the PaymentSplitter
+        /// 
+        /// This is an authenticated method which needs to be called and presented the admin badge (obtained from creating the component).
+        /// This method is used to add a shareholder to the PaymentSplitter by minting an NFT to authenticate and keep track of the shareholder
+        /// data and then airdropping it to the specific shareholder that we specified the address for.
+        /// 
+        /// # Arguments
+        /// 
+        /// * `shareholder_address` - The address of the shareholder. This address will be airdropped the NFT.
+        /// * `shareholder_shares` - The amount of shares that belong to this shareholder.
+        #[auth(admin_badge_def, keep_auth)]
+        pub fn add_shareholder(
+            &mut self,
+            shareholder_address: Address,
+            shareholder_shares: Decimal
+        ) {
+            // Creating this NFT for this specific shareholder and storing it in a bucket
+            let nft_bucket: Bucket = self.shareholder_def.mint_nft(
+                self.number_of_shareholders,
+                ShareHolder {
+                    address: shareholder_address,
+                    shares: shareholder_shares,
+                    total_xrd_owed: Decimal(0),
+                    total_xrd_withdrawn: Decimal(0)
+                },
+                auth
+            );
+            self.number_of_shareholders += 1;
+
+            // Airdropping the NFT to the user with the specified address
+            Account::from(shareholder_address).deposit(nft_bucket);
```

Comment:
We removed the Account API with v0.3. There is an example here: https://docs.radixdlt.com/main/scrypto/release_notes/v0_3.html#_removed_the_account_api on sending funds to an account component. 

But, what we recommend, is not to send directly to a wallet. Instead, return the NFT and let the caller send the token to the address they want. Calling `add_shareholder` and depositing to the account can be achieved in a single transaction with a manifest file